### PR TITLE
Add the filesystem offload doc

### PR DESCRIPTION
**Pull request by congbobo184**
Add the doc for filesystem offload, link #4403 and solve the issue #4704

### CODE REVIEW

**Comment by congbobo184** [comment]
@Jennifer88huang @Anonymitaet  PTAL:)

File: site2/docs/cookbooks-tiered-storage.md [hunk 1]
```diff
@@ -30,6 +33,14 @@ Pulsar uses multi-part objects to upload the segment data. It is possible that a
 We recommend you add a life cycle rule your bucket to expire incomplete multi-part upload after a day or two to avoid
 getting charged for incomplete uploads.
 
+## The Filesystem offloading mechanism
+
+A topic data was stored with a lot of ledgers. Offload these ledgers into filesystem, we need a path for it.
+The implementation of the Filesystem offload to generate the path for **tenant/namespace/topic/ledgerId**.
```
Comment (codelipenghui) [inline site2/docs/cookbooks-tiered-storage.md]: The path of the data is `basePath` +  /tenant/namespace/domain/topic/ +  ledgerId-UUID right?

**Comment by congbobo184** [comment]
I think clients should know the filesystem implementation use MapFile, then they can configure the MapFile's some configuration so I add the filesystem offload mechanism.

File: site2/docs/cookbooks-tiered-storage.md (hunk 1 above)
Comment (congbobo184) [reply site2/docs/cookbooks-tiered-storage.md]: yes, it should be in this way.

**Comment by sijie** [comment]
while @congbobo184 and @codelipenghui are settling down the technical details, @Jennifer88huang and @Anonymitaet we should consider reorganizing the documentation for tiered storage. we should consider adding a section called "Storage" or "Tiered Storage" at the sidebar rather than putting everything into a cookbook.

**Comment by Jennifer88huang** [comment]
@sijie Got it. Thanks for your reminder. We should consider reorganizing it. Currently, there are two "tiered storage" sections in sidebar under "Concepts and architecture" and "Cookbooks".

File: site2/docs/cookbooks-tiered-storage.md [hunk 2]
```diff
@@ -186,6 +190,59 @@ Pulsar also provides some knobs to configure the size of requests sent to GCS.
 
 In both cases, these should not be touched unless you know what you are doing.
 
+### "filesystem" Driver configuration
+
+
+#### Configure connection address
+
```
Comment (Anonymitaet) [inline site2/docs/cookbooks-tiered-storage.md]:
```suggestion
You can configure the connection address in the `broker.conf` file.
```

File: site2/docs/cookbooks-tiered-storage.md [hunk 3]
```diff
@@ -186,6 +190,59 @@ Pulsar also provides some knobs to configure the size of requests sent to GCS.
...
+```conf
+fileSystemURI="hdfs://127.0.0.1:9000"
+```
+#### Configure Hadoop profile path
+
+In the configuration file under this path, you can configure the base path and authentication and so on.
```
Comment (Anonymitaet) [inline site2/docs/cookbooks-tiered-storage.md]:
```suggestion
The configuration file is stored in the Hadoop profile path. It contains various settings, such as base path, authentication, and so on.
```

File: site2/docs/cookbooks-tiered-storage.md [hunk 4]
```diff
@@ -186,6 +190,59 @@ Pulsar also provides some knobs to configure the size of requests sent to GCS.
...
+
+```conf
+fileSystemProfilePath="../conf/filesystem_offload_core_site.xml"
+```
+
+The model for storing this topic data use `org.apache.hadoop.io.MapFile`, so we can use all of the configuration in Apache Hadoop for `org.apache.hadoop.io.MapFile`, the following example :
```
Comment (Anonymitaet) [inline site2/docs/cookbooks-tiered-storage.md]:
```suggestion
The model for storing topic data uses `org.apache.hadoop.io.MapFile`. You can use all of the configurations in `org.apache.hadoop.io.MapFile` for Hadoop.

**Example**
```

File: site2/docs/cookbooks-tiered-storage.md [hunk 5]
```diff
@@ -186,6 +190,59 @@ Pulsar also provides some knobs to configure the size of requests sent to GCS.
...
+    </property>
+    
+    <property>
+        <name>io.map.index.interval</name>
+        <value>128</value>
+    </property>
+    
+```
+
+A detailed explanation of the role of these configurations and other configurations can be found in [Filesystem Storage](http://hadoop.apache.org/).
```
Comment (Anonymitaet) [inline site2/docs/cookbooks-tiered-storage.md]:
```suggestion
For more information about the configurations in `org.apache.hadoop.io.MapFile`, see [Filesystem Storage](http://hadoop.apache.org/).
```

**Comment by sijie** [comment]
@congbobo184 can you check @Anonymitaet 's comments?

File: site2/docs/cookbooks-tiered-storage.md [hunk 6]
```diff
@@ -186,6 +190,62 @@ Pulsar also provides some knobs to configure the size of requests sent to GCS.
...
+#### Configure Hadoop profile path
+
+The configuration file is stored in the Hadoop profile path. It contains various settings, such as base path, authentication, and so on.
+
+```conf
+fileSystemProfilePath="../conf/filesystem_offload_core_site.xml"
+```
+
+The model for storing topic data uses `org.apache.hadoop.io.MapFile`. You can use all of the configurations in `org.apache.hadoop.io.MapFile` for Hadoop.
+**Example**
```
Comment (Anonymitaet) [inline site2/docs/cookbooks-tiered-storage.md]:
```suggestion

**Example**
```

Add a blank line to correct this issue.
image

**Comment by sijie** [comment]
run cpp tests

**Comment by congbobo184** [comment]
run Integration Tests

**Pull request merged**